### PR TITLE
feat: destinations conf

### PR DESCRIPTION
**Pull request by sssoja**
# Changelog

- implemented list functionality for a simple list conf 
- the user can now see existing list data and add inputs in the form of a text field or a select field
- included unit testing for state handlers and controllers 

todo #1: test implementation against destinations conf

### CODE REVIEW

File: dashboard-frontend/src/helpers/state.spec.ts [hunk 1]
```diff
@@ -77,7 +72,7 @@ describe('getFieldState', () => {
 
     const res = getFieldState(conf, localFormData);
 
-    const resValues = Object.values(res.map(f => f.value));
+    const resValues = Object.values(res.map((f: FieldState) => f.value));
```
Comment (nandanrao) [inline dashboard-frontend/src/helpers/state.spec.ts:75]: Not sure I understand what object.values does here, given that it's being passed an array?

File: dashboard-frontend/src/pages/StudyConfPage/controllers/list.ts [hunk 2]
```diff
@@ -23,11 +23,10 @@ const list = (
...
   };
 
-  const translateFieldState = (state: FieldState[]) => {
-    const button = getButton(state);
-
-    const updatedFieldState = updateGlobalState(state, translatedConf);
+  const button = fieldState && getButton(fieldState);
 
+  const translateFieldState = (fieldState: FieldState[]) => {
+    const updatedFieldState = updateGlobalState(fieldState, translatedConf);
```
[the remark this answers is not in the document]
Comment (sssoja) [reply dashboard-frontend/src/pages/StudyConfPage/controllers/list.ts:31]: Just list, you reckon this belongs in the list controller?

File: dashboard-frontend/src/helpers/state.spec.ts (hunk 1 above)
Comment (sssoja) [reply dashboard-frontend/src/helpers/state.spec.ts:75]: True it's not needed, I've removed it

**Comment by sssoja** [comment]
Decided internally to use react-hook-form for this – closing this pull request to work on #198

**Pull request closed without merging**